### PR TITLE
update run_transformer tests to default to using pytorch native UCC instead of torch_ucc

### PR DESCRIPTION
**Pull request by Fuzzkatt**
* replace HAS_TORCH_UCC check with HAS_UCC check
* add comments for failing tests running on native ucc

### CODE REVIEW

**Comment by Fuzzkatt** [comment]
Currently failing test cases: https://docs.google.com/spreadsheets/d/1q_e_egkY1T9_02W4EeNlGXuuAqKVR-PN1oDYflp-lv8/edit#gid=0

File: apex/transformer/_ucc_util.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from torch import distributed as dist
+from torch.utils import collect_env
+
+_HAS_TORCH_UCC = None
+try:
+    import torch_ucc
+    _HAS_TORCH_UCC = True
+except ImportError:
+    _HAS_TORCH_UCC = False
+HAS_UCC = dist.is_ucc_available() or _HAS_TORCH_UCC
```
Comment (crcrpar) [inline apex/transformer/_ucc_util.py:13]: could you check `dist.is_ucc_available()` first and then check `torch_ucc` if it's unavailable?

File: tests/L0/run_transformer/test_data.py [hunk 2]
```diff
@@ -14,6 +14,9 @@
 
 
 class BroadcastDataTestBase:
+    """
+    Fails on native ucc: times out
+    """
```
Comment (crcrpar) [inline tests/L0/run_transformer/test_data.py:19]: could you replace triple (double) quotes with multiple single line comments starting with "#"?

File: tests/L0/run_transformer/test_layers.py [hunk 3]
```diff
@@ -276,9 +277,19 @@ def test_row_parallel_linear(self) -> None:
     def test_row_parallel_linear_gradient_accumulation_fusion(self) -> None:
         self._row_parallel_linear_test_impl(True, False, False)
 
+    """
+    Fails in native ucc: reduction dtype: float16 not supported
+    Full error: 
+    ec_cuda_executor_persistent.c:39   cuda ec ERROR not supported reduction dtype: float16
+    reduce_scatter_knomial.c:174  TL_UCP ERROR failed to perform dt reduction
+    """
```
Comment (crcrpar) [inline tests/L0/run_transformer/test_layers.py:285]: ditto

File: tests/L0/run_transformer/test_layers.py [hunk 4]
```diff
@@ -276,9 +277,19 @@ def test_row_parallel_linear(self) -> None:
...
     def test_row_parallel_linear_gradient_accumulation_fusion_in_fp16(self) -> None:
         self._row_parallel_linear_test_impl(True, True, False)
 
+    """
+    Fails in native ucc: 
+    RuntimeError: ProcessGroup ucc does not support _reduce_scatter_base
+    """
```
Comment (crcrpar) [inline tests/L0/run_transformer/test_layers.py:292]: ditto

File: tests/L0/run_transformer/test_layers.py [hunk 5]
```diff
@@ -410,6 +421,12 @@ def test_column_parallel_linear_async(self):
     def test_column_parallel_linear_gradient_accumulation_fusion(self):
         self._column_parallel_linear_test_impl(False, True, False, False)
 
+    """
+    Fails on native ucc: reduction dtype: float16 not supported
+    Full error: 
+    ec_cuda_executor_persistent.c:39   cuda ec ERROR not supported reduction dtype: float16
+    reduce_scatter_knomial.c:174  TL_UCP ERROR failed to perform dt reduction
+    """
```
Comment (crcrpar) [inline tests/L0/run_transformer/test_layers.py:429]: ditto

**Review by crcrpar — COMMENTED** [review]
there're some nit-pickings but lgtm overall

File: apex/transformer/_ucc_util.py (hunk 1 above)
Comment (Fuzzkatt) [reply apex/transformer/_ucc_util.py]: Added

File: tests/L0/run_transformer/test_data.py (hunk 2 above)
Comment (Fuzzkatt) [reply tests/L0/run_transformer/test_data.py]: Fixed

File: tests/L0/run_transformer/test_layers.py (hunk 3 above)
Comment (Fuzzkatt) [reply tests/L0/run_transformer/test_layers.py]: Fixed

File: tests/L0/run_transformer/test_layers.py (hunk 4 above)
Comment (Fuzzkatt) [reply tests/L0/run_transformer/test_layers.py]: Fixed

File: tests/L0/run_transformer/test_layers.py (hunk 5 above)
Comment (Fuzzkatt) [reply tests/L0/run_transformer/test_layers.py]: Fixed

File: apex/transformer/_ucc_util.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+from packaging.version import Version, parse
+
+import torch
+from torch import distributed as dist
+from torch.utils import collect_env
```
[the remark this answers is not in the document]
Comment (Fuzzkatt) [reply apex/transformer/_ucc_util.py]: Fixed

**Pull request merged**